### PR TITLE
Added promoting track tasks

### PR DESCRIPTION
**Pull request by shiraji**
see #103 

Feel free to comment on this pull request. I will remove [WIP] when this pull request is ready.

These commit will be rebase anytime I think it's necessary.

### CODE REVIEW

**Comment by shiraji** [comment]
OK, now I can promote the track from alpha to beta. 

### TODO

* handling multiple versions
* handling conflicts (if there is already beta version, what happen...?)
* creating actual tasks for promoting

**Comment by shiraji** [comment]
I changed my mind and created following tasks:

* promote${variationName}AlphaToBeta
* promote${variationName}AlphaToProduction
* promote${variationName}BetaToProduction

Does these tasks violate what this library designs for? if not, could you check my pull request, @bhurling ? Thanks.

**Comment by WonderCsabo** [comment]
I would love to see this merged. :)

**Comment by VerachadW** [comment]
+1

**Comment by rocboronat** [comment]
+1

**Comment by julioz** [comment]
@shiraji This looks great, and I would also love to see this merged, but could I suggest you add to your pull request a change to the `README.md` file regarding the new `Task`s and how to use them (specially the `promoteToProd` task when setting a `userFraction` to perform a staged rollout).

**Comment by filipesperandio** [comment]
Any plans for merging this in?

File: src/main/groovy/de/triplet/gradle/play/PlayPublishTask.groovy [hunk 1]
```diff
@@ -48,4 +50,14 @@ class PlayPublishTask extends DefaultTask {
         }
     }
 
+    def getTrackByName(String trackName) {
```
Comment (ChristianBecker) [inline src/main/groovy/de/triplet/gradle/play/PlayPublishTask.groovy]: This method can be written in more Groovy-ish way:
```
    def getTrackByName(String trackName) {
        def tracksList = edits.tracks().list(variant.applicationId, editId).execute()
        return tracksList.getTracks().find { it.getTrack() == trackName }
    }
```

Comment (shiraji) [inline src/main/groovy/de/triplet/gradle/play/PlayPublishTask.groovy]: see https://github.com/Triple-T/gradle-play-publisher/pull/110#discussion_r106444123

**Comment by paour** [comment]
Thanks! I've merged this very useful PR into [my fork](https://github.com/paour/gradle-play-publisher) – it's really too bad that the author of this project is letting some PRs languish!

**Comment by rangelier** [comment]
I'm currently working on a project for automating deployments. I'm unable to find documentation in the current API that involves promoting alpha -> beta -> production. Is this possible?

**Comment by shiraji** [comment]
@SUPERCILEX 
According to what you write https://github.com/Triple-T/gradle-play-publisher/issues/104#issuecomment-391416685 , I think this PR is useless, right? Why don't you close this and #110

**Comment by SUPERCILEX** [comment]
@shiraji Oh, yeah, I guess so. 😢 I see this PR has been open for almost 3 years—really sorry about that. I'm joining the team to try and get things in order, but I've been holding off on code and PRs until I have the community related stuff tidied up first.

To give you an update of what's going on, we're moving the plugin to Kotlin which is basically going to kill all PRs anyway. After that, if you get the green light on an issue, I promise we'll do our best to make sure your work doesn't go to waste. So again, my most sincere apologies for having to let this PR go. 😊

**Pull request closed without merging**